src/java/edu/slu/action/ObjectAction.java

```diff
@@ -1033,91 +1139,95 @@ else if(isReleased){
-                                updatedObject.remove(key);
-                                updatedObject.append(key, received.get(key));
-                                updateCount +=1 ;
+            if(received.containsKey("@id")){
+                String updateHistoryNextID = received.getString("@id");
+                query.append("@id", updateHistoryNextID);
+                BasicDBObject originalObject = (BasicDBObject) mongoDBService.findOneByExample(Constant.COLLECTION_ANNOTATION, query); //The originalObject DB object
+                BasicDBObject updatedObject = (BasicDBObject) originalObject.copy(); //A copy of the original, this will be saved as a new object.  Make all edits to this variable.
+                boolean alreadyDeleted = checkIfDeleted(JSONObject.fromObject(originalObject));
+                boolean isReleased = checkIfReleased(JSONObject.fromObject(originalObject));
+                if(alreadyDeleted){
+                    writeErrorResponse("The object you are trying to update is deleted.", HttpServletResponse.SC_BAD_REQUEST);
+                }
+                else if(isReleased){
+                    writeErrorResponse("The object you are trying to update is released.  Fork to make changes.", HttpServletResponse.SC_BAD_REQUEST);
+                }
+                else{
+                    if(null != originalObject){
+                        Set<String> update_anno_keys = received.keySet();
+                        boolean triedToSet = false;
+                        int updateCount = 0;
+                        //If the object already in the database contains the key found from the object recieved from the user, update it barring a few special keys
+                        //Users cannot update the __rerum property, so we ignore any update action to that particular field.  
+                        for(String key : update_anno_keys){
+                            if(originalObject.containsKey(key) ){
+                                //Skip keys we want to ignore and keys that match but have matching values
+                                if(!(key.equals("@id") || key.equals("__rerum") || key.equals("objectID") || key.equals("_id")) && received.get(key) != originalObject.get(key)){
+                                    updatedObject.remove(key);
+                                    updatedObject.append(key, received.get(key));
+                                    updateCount +=1 ;
+                                }
+                            }
+                            else{
+                                triedToSet = true;
+                               // break;
                             }
                         }
-                        else{
-                            triedToSet = true;
-                            break;
+                        if(triedToSet){
+                            //@cubap @theHabes We continued with what we could patch.  Do we tell the user at all?
```

Comment:
Maybe the response can be something like a list of keys with results? I'm not sure what to do with this.